### DISCUSSION TITLE
Problem createSprite function

### DISCUSSION THREAD

**Discussion by FrancescoIoppolo** [Bodmer/TFT_eSPI#1850] [General]
I have a problem, I am trying to create a progress bar but it does not show if the completion is not 100%
The problem is here:
  
  **sprite_bargraph.createSprite(200, 22);
    sprite_bargraph.drawRoundRect(0, 0, 200, 16, 4, TFT_WHITE);
    sprite_bargraph.fillRoundRect(1, 1, 10, 14, 4, TFT_WHITE);
    sprite_bargraph.pushSprite(60, 110);**

Everything else works fine!

this only happens with the Firebase callback function. With other calback functions it works fine

do you have any ideas?

this is the function

Firebase.Storage.downloadOTA(&fbdo, STORAGE_BUCKET_ID /* Firebase Storage bucket id */, "update.bin" /* path of firmware file stored in the bucket */, downloadBinFirebase /* callback function */)

this is my code

```
#include <TFT_eSPI.h>
#include <SPI.h>
TFT_eSPI tft = TFT_eSPI();

TFT_eSprite sprite_bargraph = TFT_eSprite(&tft);


void downloadBinFirebase(FCS_DownloadStatusInfo info)
{
  if (info.status == fb_esp_fcs_download_status_init)
  {
    Serial.printf("Downloading firmware %s (%d)\n", info.remoteFileName.c_str(), info.fileSize);
  }
  else if (info.status == fb_esp_fcs_download_status_download)
  {
    Serial.printf("Downloaded %d%s\n", info.progress, "%");

    int percent = info.progress;

    String myStringPercent = String(percent);

    myStringPercent = myStringPercent + "%";

    tft.setTextColor(TFT_WHITE, TFT_BLACK);
    tft.setTextSize(1);
    tft.drawString("UPDATE LAST VERSION" , 108,  90);
    tft.drawString(myStringPercent, 150,  145);
    tft.drawString("! Attention do not disconnect the device !", 40,  160);
        
    sprite_bargraph.createSprite(200, 22);
    sprite_bargraph.drawRoundRect(0, 0, 100, 16, 4, TFT_WHITE);
    sprite_bargraph.fillRoundRect(1, 1, percent, 14, 4, TFT_WHITE);
    sprite_bargraph.pushSprite(60, 110);    
  }
  else if (info.status == fb_esp_fcs_download_status_complete)
  {
    Serial.println("Update firmware completed.");
    Serial.println();
    Serial.println("Restarting...\n\n");
    delay(2000);
    ESP.restart();
  }
  else if (info.status == fb_esp_fcs_download_status_error)
  {
    Serial.printf("Download firmware failed, %s\n", info.errorMsg.c_str());
  }
}
```
This is while loading:
![IMG20220526081608](https://user-images.githubusercontent.com/94544242/170429829-3185e922-b673-42dc-b4c4-17e5930210bf.jpg)

This is at the end of the upload:
![IMG20220526082250](https://user-images.githubusercontent.com/94544242/170429843-584c57c6-4ac0-4a52-bf48-07f0356911a6.jpg)

**Comment by Bodmer** [comment]
This suggests that there is insufficient memory to create the sprite until some other activity is comeplete and then sufficient memory becomes available. The sprite will require 8800 bytes. the creatSprite function returns a null pointer if the sprite cannot be created so that is a simple check. There is no clue to which processor is being used but the ESP8266 can easily run out of memory for web based interaction.

It is also odd that a sprite 200 pixels wide is created, the drawn rectangles are 100 pixels wide, yet the final image shows that the sprite appears to be filled with a rectangle 200 pixels wide! Is there an explanation for that?

Unless a complete minimal sketch is provided that I can run quickly and easliy with no special support code then I will not be able to help further.
